client/src/views/workspaces/WorkspaceSharingModal.vue

```diff
@@ -191,11 +267,73 @@ function isSelectDisabled(role: WorkspaceRole | null): boolean {
   return false;
 }
 
+async function onBatchSharingActivate(): Promise<void> {
+  if (showCheckboxes.value === true) {
+    for (const user of userRoles.value) {
+      user.isSelected = false;
+    }
+    membersCheckbox.value = false;
+  }
+  showCheckboxes.value = !showCheckboxes.value;
+}
+
+function getLowestSelectedProfile(): UserProfile {
+  let lowestProfile = UserProfile.Admin;
+  for (const user of selectedUsers.value) {
+    if (user.role[0].profile === UserProfile.Outsider) {
+      return UserProfile.Outsider;
+    }
+    if (user.role[0].profile === UserProfile.Standard) {
+      lowestProfile = UserProfile.Standard;
+    }
+  }
+  return lowestProfile;
+}
+
+function getHighestSelectedRole(): WorkspaceRole {
+  let highestRole = WorkspaceRole.Reader;
+  for (const user of selectedUsers.value) {
+    if (!user.role[1]) {
+      continue;
+    }
+    if (user.role[1] in [WorkspaceRole.Manager, WorkspaceRole.Owner]) return user.role[1];
```

Comment:
I don't think you can use `in` here, it doesn't work as in Python.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Expressions_and_operators#in
It's used to check if an object has a property and in this case, if an array as an index (not a value).

```
> a = [2,3,4,5];
> 4 in a;
false
> 2 in a;
true
```